src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -217,23 +217,16 @@ public static JaegerSpanRef ToJaegerSpanRef(this in ActivityLink link)
 
         public static JaegerTag ToJaegerTag(this KeyValuePair<string, object> attribute)
         {
-            switch (attribute.Value)
+            return attribute.Value switch
             {
-                case string s:
-                    return new JaegerTag(attribute.Key, JaegerTagType.STRING, vStr: s);
-                case int i:
-                    return new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: Convert.ToInt64(i));
-                case long l:
-                    return new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: l);
-                case float f:
-                    return new JaegerTag(attribute.Key, JaegerTagType.DOUBLE, vDouble: Convert.ToDouble(f));
-                case double d:
-                    return new JaegerTag(attribute.Key, JaegerTagType.DOUBLE, vDouble: d);
-                case bool b:
-                    return new JaegerTag(attribute.Key, JaegerTagType.BOOL, vBool: b);
-            }
-
-            return new JaegerTag(attribute.Key, JaegerTagType.STRING, vStr: attribute.Value.ToString());
+                string s => new JaegerTag(attribute.Key, JaegerTagType.STRING, vStr: s),
```

Comment:
we now need to deal with the array of primitives since https://github.com/open-telemetry/opentelemetry-dotnet/pull/999 is merged.